client.go

```diff
@@ -593,6 +593,16 @@ func (c *Client) Truncate(path string, size int64) error {
 	return c.setstat(path, sshFileXferAttrSize, uint64(size))
 }
 
+// SetExtendedAttrs sets the extended attributes of the named file.
+func (c *Client) SetExtendedAttrs(path string, extended []StatExtended) error {
```

Comment:
It seems like the common abbreviation for “extended attributes” is `xattr` So, that would suggest `SetXattr`, which would match the `setxattr` call in OpenSSH. But I don’t think we want our name that short. `SetExtAttrs()` might be a good inbetween? But I’m still fine with this longer name as well.

Maybe we could also implement this as `SetExtAttrs(path string, xattrs ...StatExtended) error` … though checking it on go playground, it seems this doesn’t allow for the type-elided shorthand `cl.SetExtAttr(path, { ExtType: "foo", ExtData: "bar" })` to be used. 🙁  Meanwhile, `cl.SetExtAttrs(path, []sftp.StatExtended{ { ExtType: "foo", ExtData: "bar" } })` only needs the one `sftp.StatExtended` type name to be used for any number of xattrs.

A common API design that some have picked is `SetExtAttrs(path string, xattrs ...string)` and then provide alternating `type` and `data` values that get turned into the `[]sftp.StatExtended` internally. But I’m not a big fan of this easy-to-misalign approach. But then again, it is integrated into the standard `slog` library, so… 🤷‍♀️ 

In the alternative, we could provide two functions: `SetExtAttrs(path string, xattrs []StatExtended)` and `SetExtAttr(path, typ, data string)` thus anyone wanting to set a single attribute can avoid the `StatExtended` altogether, and then for more than one, they can make use of the `[]StatExtended{ {}, {} }` type elision.

Reply:
Following the [draft RFCs](https://datatracker.ietf.org/doc/html/draft-ietf-secsh-filexfer-13#section-7.14) the extended attributes in the SFTP protocol do not necessarily match to extended atributes on the system, so I would avoid the very short `xattr` to avoid confusion. Maybe `SetExtendedData` is a better name to correspond with [the FileInfoExtendedData interface](https://github.com/pkg/sftp/blob/637088883317ec6aced847f1f5d5b007354e0145/attrs.go#L96-L99) ?

Another catch is that it is very unspecified what `setstat` actually does with extended attributes. It might, in case of real extended attributes on files, just replace all current attributes.

Reply:
Second checked the standard, and the field is called `extended_data`, so 👍